examples/explore_singular.c

```diff
@@ -0,0 +1,152 @@
+//
+// Created by Mikael Vejdemo Johansson on 4/7/15.
+//
+// Intention: read in a parser, generate the system of equations for its
+// generating functions
+//
+
+#include <inttypes.h>
+#include "../src/backends/contextfree.h"
+#include "../src/backends/lr.h"
+#include <stdio.h>
+
+static const char *nonterminal_name(const HCFGrammar *g, const HCFChoice *nt)
+{
+  static char buf[16] = {0}; // 14 characters in base 26 are enough for 64 bits
+
+  // find nt's number in g
+  size_t n = (uintptr_t)h_hashtable_get(g->nts, nt);
+
+  // NB the start symbol (number 0) is always "A".
+  int i;
+  for(i=14; i>=0 && (n>0 || i==14); i--) {
```

Comment:
Example code should be written for clarity over conciseness, cleverness, and performance. Could you expand on what this loop is doing? In comments is fine, but making the code clearer is ideal.